### ISSUE TITLE
Test for `FORCE_HYPERLINK` gives a warnings during `omz update`

### ISSUE THREAD

**Issue by whxvd** [ohmyzsh/ohmyzsh#11838]
### Describe the bug

Running `omz update` in some cases gives two warnings:

```
supports_hyperlinks:2: FORCE_HYPERLINK: parameter not set
supports_hyperlinks:2: FORCE_HYPERLINK: parameter not set
```

### Steps to reproduce

It seems to always happen, when there actually is something to update, which can be forced with running `git reset --hard HEAD^` before running `omz update`.

### Expected behavior

No such warnings should happen.

### OS / Linux distribution

Arch Linux

### Zsh version

5.9

### Terminal emulator

Alacritty

### If using WSL on Windows, which version of WSL

None

### Additional context

I suspect that it is because of the tests `[ -n "$FORCE_HYPERLINK" ]` in the code base. That probably should be `[[ -v FORCE_HYPERLINK ]]`. Or `[ -n "${FORCE_HYPERLINK:-}" ]`.

**Comment by mcornella** [comment]
Do you have `set +u` or `setopt nounset` somewhere in your configurations? It might be in `/etc/zshenv` or `$HOME/.zshenv`. If so, I already know how to solve it.

**Comment by carlosala** [comment]
Thanks for your testing and the report!

**Comment by whxvd** [comment]
> You can test the fix with `omz pr test 11856`.

I did not find a way to really test the PR. When I run

``` shell
omz pr test 11856
git reset --hard HEAD^
omz update
```

I get the message that OMZ already is at the latest version, probably because `omz update` does not look at the PR branch. I tried to test by manually introducing the change while being on the `master` branch, but then `omz update` temporarily stashes away the change.

I did not try further. But I consider it highly unlikely that the PR not solves the issue or introduces new problems.

On the other hand, I do not know whether the PR really is necessary. I rather consider my previous use of `$HOME/.zshenv` (including twiddling with shell options) a misuse and the real issue. However, merging the PR almost certainly will not hurt, either.

All the best!
